src/configuration/index.js

```diff
@@ -0,0 +1,205 @@
+import Promise from 'pinkie';
+import { fsObjectExists, readFile } from '../utils/promisified-functions';
+import Option from './option';
+import optionSource from './option-source';
+import { cloneDeep, castArray } from 'lodash';
+import { ensureOptionValue as ensureSslOptionValue } from '../utils/parse-ssl-options';
+import OPTION_NAMES from './option-names';
+import { optionValueToRegExp } from './option-conversion';
+import createFilterFn from '../utils/create-filter-fn';
+import resolvePathRelativelyCwd from '../utils/resolve-path-relatively-cwd';
+
+const CONFIGURATION_FILENAME = '.testcaferc.json';
+
+const ERR_READ_CONFIG_FILE                 = 'An error is occurred during reading the configuration filename.';
+const ERR_CONFIG_FILE_IS_NOT_WELLFORMATTED = 'Configuration filename is not well-formatted.';
+
+const STATIC_CONTENT_CACHING_SETTINGS = {
+    maxAge:         3600,
+    mustRevalidate: false
+};
+
+export default class Configuration {
+    constructor () {
+        this._options  = {};
+        this._filePath = resolvePathRelativelyCwd(CONFIGURATION_FILENAME);
+    }
+
+    static _fromObj (obj) {
+        const result = Object.create(null);
+
+        Object.entries(obj).forEach(entry => {
+            const key    = entry[0];
+            const value  = entry[1];
+            const option = new Option(key, value);
+
+            result[key] = option;
+        });
+
+        return result;
+    }
+
+    async load () {
+        if (!await fsObjectExists(this.filePath))
+            return;
+
+        let configurationFileContent = null;
+
+        try {
+            configurationFileContent = await readFile(this.filePath);
+        }
+        catch (e) {
+            console.log(ERR_READ_CONFIG_FILE); // eslint-disable-line no-console
+        }
+
+        try {
+            const optionsObj = JSON.parse(configurationFileContent);
```

Comment:
It makes sense to use [JSON5](https://www.npmjs.com/package/json5) rather than standard JSON: it has relatively relaxed grammar and allows code comments which can be quite handy sometimes. AFAIK many popular tools use it.

Also, there is a trend recently to use JS files that export configuration object instead of dealing with all the JSON-related shenanigans. Thus, you can have arbitrary configuration logic that depends on environment, for example.

Reply:
I glad to see you @inikulin . We've discussed about other configuration file formats (`js` and `yml`) and decided to support only the `JSON` format in the first implementation.

Thank you for the provided information  about `JSON5`. But I think we will use current `JSON` implementation until there are problems with it.